### PR TITLE
MS Teams - Adding ring-user command - reopened

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22329

## Description
continuing https://github.com/demisto/content/pull/6143 after corruption...
Added ring user command and added micrisoft-teams suffix to commands

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 1]
```diff
@@ -1495,6 +1495,74 @@ def messages() -> Response:
...
+        "requestedModalities": [
+            "audio"
+        ],
+        "mediaConfig": {
+            "@odata.type": "#microsoft.graph.serviceHostedMediaConfig",
+        },
+        "tenantId": tenant_id
+    }
+    response = http_request(method='POST',
+                            url='https://graph.microsoft.com/v1.0/communications/calls',
```
Comment (Itay4) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]:
```suggestion
                            url=f'{GRAPH_BASE_URL}/v1.0/communications/calls',
```

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 2]
```diff
@@ -1495,6 +1495,74 @@ def messages() -> Response:
...
+    """Rings a user on Teams.
+
+    Notes:
+        This is a ring only! no media plays in case the generated call is answered.
+
+    Returns:
+        None.
+    """
+    bot_id = demisto.params().get('bot_id')
+    tenant_id = demisto.args().get('tenant_id')
```
Comment (Itay4) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]: this shouldnt be an argument.. take it from `integrationContext`

File: Integrations/MicrosoftTeams/MicrosoftTeams.yml [hunk 3]
```diff
@@ -221,11 +221,83 @@ script:
...
+      required: true
+      secret: false
+    - default: false
+      description: The tenant ID for the Teams.
+      isArray: false
+      name: tenant_id
+      required: true
+      secret: false
+    - default: false
+      description: The callback URI for the call for the bot who will recive a callback
```
Comment (Itay4) [inline Integrations/MicrosoftTeams/MicrosoftTeams.yml]: it is still not clear what should be passed here..

File: Integrations/MicrosoftTeams/MicrosoftTeams.py (hunk 2 above)
Comment (roysagi) [reply Integrations/MicrosoftTeams/MicrosoftTeams.py]: done

File: Integrations/MicrosoftTeams/MicrosoftTeams.py (hunk 1 above)
Comment (roysagi) [reply Integrations/MicrosoftTeams/MicrosoftTeams.py]: done

File: Integrations/MicrosoftTeams/MicrosoftTeams.yml (hunk 3 above)
Comment (roysagi) [reply Integrations/MicrosoftTeams/MicrosoftTeams.yml]: I'll add it to the integration description - it might be too much for an argument description

**Comment by roysagi** [comment]
@kirbles19 I'll create a new README but we need to edit some additional info on the docs that can be found here: https://support.demisto.com/hc/en-us/articles/360034095994-Microsoft-Teams

Most notably how to set up calling permissions to the bot.

File: Integrations/MicrosoftTeams/MicrosoftTeams.yml (hunk 3 above)
Comment (Itay4) [reply Integrations/MicrosoftTeams/MicrosoftTeams.yml]: dont think integration description is the place for that..

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 4]
```diff
@@ -1495,6 +1495,80 @@ def messages() -> Response:
...
+    username_to_call = demisto.args().get('username')
+    users: list = get_users()
+    user_id: str = str()
+    found_member: bool = False
+    for user in users:
+        if username_to_call in {user.get('displayName', ''), user.get('mail'), user.get('userPrincipalName')}:
+            found_member = True
+            user_id = user.get('id', '')
+            break
+    if not found_member:
```
Comment (Itay4) [inline Integrations/MicrosoftTeams/MicrosoftTeams.py]: you dont need that flag, just check `if user_id:`

Comment (roysagi) [reply Integrations/MicrosoftTeams/MicrosoftTeams.py]: done

**Pull request closed without merging**